apps/web/src/components/Shared/Embed/Space/index.tsx

```diff
@@ -0,0 +1,133 @@
+import SmallUserProfile from '@components/Shared/SmallUserProfile';
+import { Icons } from '@components/Spaces/Common/assets/Icons';
+import { MicrophoneIcon } from '@heroicons/react/outline';
+import { getLensAccessToken, getLensMessage } from '@huddle01/auth';
+import type { Profile, Publication } from '@lenster/lens';
+import { useProfilesQuery } from '@lenster/lens';
+import getPublicationAttribute from '@lenster/lib/getPublicationAttribute';
+import type { SpaceMetadata } from '@lenster/types/spaces';
+import { Button, Spinner } from '@lenster/ui';
+import clsx from 'clsx';
+import { type FC } from 'react';
+import { useSpacesStore } from 'src/store/spaces';
+import { useAccount, useSignMessage } from 'wagmi';
+
+import Wrapper from '../Wrapper';
+
+interface SpaceProps {
+  publication: Publication;
+}
+
+const Space: FC<SpaceProps> = ({ publication }) => {
+  const { address } = useAccount();
+  const { metadata } = publication;
+
+  const { setShowSpacesLobby, setLensAccessToken, lensAccessToken, setSpace } =
+    useSpacesStore();
+
+  const space: SpaceMetadata = JSON.parse(
+    getPublicationAttribute(metadata.attributes, 'audioSpace')
+  );
+
+  const { signMessage, isLoading: signing } = useSignMessage({
+    onSuccess: async (data) => {
+      const token = await getLensAccessToken(data, address as string);
+      if (token.accessToken) {
+        setShowSpacesLobby(true);
+        setLensAccessToken(token.accessToken);
+        setSpace({
+          ...space,
+          title: metadata.content
+        });
+      }
+    }
+  });
+
+  const { data, loading } = useProfilesQuery({
+    variables: {
+      request: { ownedBy: [space.host] }
+    }
+  });
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  const hostProfile = data?.profiles?.items?.find(
+    (profile) => profile?.ownedBy === space.host
+  ) as Profile;
+
+  const calculateRemainingTime = () => {
+    const now = new Date();
+    const targetTime = new Date(space.startTime);
+    const timeDifference = targetTime.getTime() - now.getTime();
+    if (timeDifference <= 0) {
+      return 'Start Listening';
+    }
+    const hours = Math.floor(timeDifference / (1000 * 60 * 60));
```

Comment:
do we not have utils for this @bigint?

Reply:
we already have formatTime util using dayjs @vrajdesai78

Reply:
I checked that, but it won't work for this case, bcz it just returns string in days, and I want it in a different format.

Reply:
What will be the exact output looks like? 🙇🏼

Reply:
![image](https://github.com/lensterxyz/lenster/assets/43074241/107dbfc3-ccc9-419e-9c9a-2ab23f9dcc9d)

Reply:
imo this can be done using dayjs, add util there